### PR TITLE
Database health-check, authorization with github and routes

### PR DESCRIPTION
**Pull request by KacperMalachowski**
I changed the name of register endpoint compared to documentation to /login/callback because of that name better describes this endpoint.

### CODE REVIEW

File: src/config/development.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  mongo: {
+    connectionString: "mongodb://localhost:27017",
+    options: {
+      useNewUrlParser: true,
+      useUnifiedTopology: true
+    }
+  },
+  serverOptions: {
+    port: process.env.PORT | 3000,
```
Comment (hayuna) [inline src/config/development.js]: Screenshot 2020-08-21 at 15 43 12
Is this expected value? I think you missed or operator with bitwise operator

File: src/config/development.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  },
+  githubOAuth: {
+    clientID: process.env.GITHUB_CLIENTID,
+    clientSecret: process.env.GITHUB_CLIENT_SECRET,
+    callbackURL: "https://127.0.0.1/login/callback"
```
Comment (hayuna) [inline src/config/development.js]: Can we store callback into .env?

File: src/controllers/AuthController.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const AuthController = {
+  redirectedUser: (req, res) => {
```
Comment (hayuna) [inline src/controllers/AuthController.js]: This is action unfinished. Should be named like `redirectUser`

File: src/controllers/HealthCheckController.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+const serviceStatus = {
+  database: {},
+  service: "Ready"
+}
+
+const HealthCheckController = {
+
+  getServiceStatus: () => {
+    switch (mongoose.connection.readyState) {
```
Comment (hayuna) [inline src/controllers/HealthCheckController.js]: Cant we just return this status?

File: src/models/User.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+const { Schema, model } = require('mongoose');
+
+const UserSchema = new Schema({
+  id: {
```
Comment (hayuna) [inline src/models/User.js]: mongo generate their own id named `_id` and this is unique by default. I think, we can remove this field

**Review by hayuna — CHANGES_REQUESTED** [review]
This PR looks chaotic. Please talk with your PM about some convention in naming commits, naming branches and naming pull requests.

**Comment by hayuna** [comment]
@bwieckow If we need to change 15 files to finish task, it looks like this task should be split to lower ones.

File: src/config/development.js (hunk 2 above)
Comment (Przemocny) [reply src/config/development.js]: it should be from process.env too

File: src/controllers/AuthController.js (hunk 3 above)
Comment (Przemocny) [reply src/controllers/AuthController.js]: this action has not suitable name - it isnt redirectUser, its more like redirectToIndexWithUser for me

File: src/controllers/HealthCheckController.js (hunk 4 above)
Comment (Przemocny) [reply src/controllers/HealthCheckController.js]: this.getServiceStatus should return { isConnected, numericStatus, msg } to have all informations about status

File: src/index.js [hunk 6]
```diff
@@ -1,18 +1,23 @@
 const express = require('express');
 const bodyParser = require('body-parser');
```
Comment (Przemocny) [inline src/index.js]: bodyparser exist in express

app.use(express.json()) - .urlencoded is probably there too

File: src/index.js [hunk 7]
```diff
@@ -1,18 +1,23 @@
...
+const passport = require('passport');
 const config = require('./config');
 const routes = require('./routes');
 const mongo = require('./mongodb');
+require('./config/passport');
+
+console.log(config.serverOptions.port)
 
 mongo.connect(config.mongo.connectionString, config.mongo.options);
```
Comment (Przemocny) [inline src/index.js]: from here this is onInit(app=express()){} function, down to the app.listen

File: src/routes/auth.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+const router = require('express').Router();
+const passport = require('passport');
+const AuthController = require('../controllers/AuthController');
+
+router.get('/callback', passport.authenticate('github'), AuthController.redirectedUser)
+
+router.get('/', passport.authenticate('github'));
```
Comment (Przemocny) [inline src/routes/auth.js]: / should be first

/callback should be second route

File: src/config/passport.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+const passport = require('passport');
```
Comment (Przemocny) [inline src/config/passport.js:1]: Why env doesnt have imports and exports ? 
and why not typescript ?

probably matching config of env is here https://github.com/Przemocny/ts_env_scripts

File: src/models/User.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    type: String,
+    unique: true
+  },
+  username: String,
+  token: String
+});
+
+UserSchema.statics.findOneOrCreate = function (accessToken, profile, cb)  {
```
Comment (Przemocny) [inline src/models/User.js]: this should be async/await promise, not callback based method

**Review by Przemocny — CHANGES_REQUESTED** [review]
3/10 imo, to fix

File: src/index.js (hunk 6 above)
Comment (hayuna) [reply src/index.js]: >>app.use(express.json()) - .urlencoded is probably there too

Correct

**Review by hayuna — CHANGES_REQUESTED** [review]
Comments above

File: src/config/development.js (hunk 1 above)
Comment (KacperMalachowski) [reply src/config/development.js]: Yeah, I missed or operator with bitwise operator, fixed now.

File: src/config/development.js (hunk 2 above)
Comment (KacperMalachowski) [reply src/config/development.js]: Ok, I added dotenv package and configure it in src/config/index.js and I added example .env file to repository.

File: src/controllers/AuthController.js (hunk 3 above)
Comment (KacperMalachowski) [reply src/controllers/AuthController.js]: Auth controller and redirectedUser replaced with buil-in passport.js redirect option.

File: src/controllers/HealthCheckController.js (hunk 4 above)
Comment (KacperMalachowski) [reply src/controllers/HealthCheckController.js]: Added more informative object with isConnected (bool for DB) or isReady (bool for service) and numeric status and message for both DB and Service.
Current structure:
`status {
  database {
     isConnected: bool,
     status: integer,
     message: string
  },
  service {
     isReady: bool,
     status: integer,
     message: string
  }
}`

File: src/models/User.js (hunk 5 above)
Comment (KacperMalachowski) [reply src/models/User.js]: This is id  from github, I renamed it to githubId now. We use it to check if user already exists in our database.

File: src/models/User.js (hunk 10 above)
Comment (KacperMalachowski) [reply src/models/User.js]: Removed callback and convert to async/await based.

**Comment by KacperMalachowski** [comment]
isReady function created for feature better service health check (which will be using more metrics than DB connection), but now would be replaced with simple invoke isCOnnected

File: .eslintrc [hunk 11]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (hayuna) [inline .eslintrc:1]: You can use our config if you have some problems

**Pull request closed without merging**